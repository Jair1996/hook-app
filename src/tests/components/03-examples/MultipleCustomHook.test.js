import { shallow } from "enzyme";
import { MultipleCustomHook } from "../../../components/03-examples/MultipleCustomHook";
import { useCounter } from "../../../hooks/useCounter";
import { useFetch } from "../../../hooks/useFetch";
//Implementacion falsa, evita las llamadas al useFetch
jest.mock("../../../hooks/useFetch");
jest.mock("../../../hooks/useCounter");

describe("Pruebas en <MultipleCustomHook />", () => {
  beforeEach(() => {
    useCounter.mockReturnValue({
      counter: 10,
      increment: () => {},
    });
  });

  test("debe de mostrarse correctamente", () => {
    //Usamos esto porque nos arrojara un error en el componente ya que no podra
    //llamar al useFetch, ya que hicimos un mock
    useFetch.mockReturnValue({
      data: null,
      loading: true,
      error: null,
    });

    const wrapper = shallow(<MultipleCustomHook />);

    expect(wrapper).toMatchSnapshot();
  });

  test("debe de mostrar la informaciÃ³n", () => {
    useFetch.mockReturnValue({
      data: [
        {
          author: "Jair",
          quote: "Hola Mundo",
        },
      ],
      loading: false,
      error: null,
    });

    const wrapper = shallow(<MultipleCustomHook />);

    expect(wrapper.find(".alert").exists()).toBe(false);
    expect(wrapper.find(".mb-0").text().trim()).toBe("Hola Mundo");
    expect(wrapper.find("footer").text().trim()).toBe("Jair");
    console.log(wrapper.html());
  });
});
