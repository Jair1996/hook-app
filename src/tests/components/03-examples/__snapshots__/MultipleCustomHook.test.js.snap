// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <MultipleCustomHook /> debe de mostrarse correctamente 1`] = `
<div>
  <h1>
    BreakingBad quotes
  </h1>
  <hr />
  <div
    className="alert alert-info"
  >
    Loading...
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Siguiente frase
  </button>
</div>
`;
